apix/parsers/openapi.py

```diff
@@ -0,0 +1,298 @@
+        """Extract body parameters for Swagger 2.0 style specs"""
+        params = []
+        for param in operation.get("parameters", []):
+            if param.get("in") != "body" or "schema" not in param:
+                continue
+            schema = param["schema"]
+            if "properties" not in schema:
+                continue
+            required_fields = set(schema.get("required", []))
+            for prop_name, prop_schema in schema["properties"].items():
+                status = "required" if prop_name in required_fields else "optional"
+                prop_type = prop_schema.get("type", "any")
+                params.append(f"{prop_name} ~ {status} ~ {prop_type}")
+        return params
+
+    def _add_or_update_method(self, entity, method_name, formatted_path, params):
+        """Add a new method or update an existing one with additional paths/params.
+
+        Args:
+            entity: The entity name (e.g., 'pets', 'users')
+            method_name: The method/operation name (e.g., 'createPet', 'listUsers')
+            formatted_path: HTTP method and path (e.g., 'GET /pets')
+            params: List of parameter strings in format 'name ~ status ~ type'
+        """
+        for existing_method in self._data[entity]["methods"]:
+            if method_name in existing_method:
+                existing_method[method_name]["paths"].append(formatted_path)
+                # Use set for O(1) lookup of existing params
+                existing_params = set(existing_method[method_name].get("params", []))
+                new_params = [p for p in params if p not in existing_params]
+                existing_method[method_name]["params"].extend(new_params)
+                return
+
+        self._data[entity]["methods"].append(
+            {method_name: {"paths": [formatted_path], "params": params}}
+        )
+
+    def _parse_openapi_spec(self, spec):
+        """Parse an OpenAPI specification dictionary"""
+        paths = spec.get("paths", {})
+
+        for path, path_item in paths.items():
+            for http_method in ["get", "post", "put", "patch", "delete", "head", "options"]:
+                if http_method not in path_item:
+                    continue
+
+                operation = path_item[http_method]
+                tags = operation.get("tags", [])
+                operation_id = operation.get("operationId")
+
+                entity = self._extract_entity_from_path(path, tags)
+                method_name = self._extract_method_name(operation_id, http_method, path)
+
+                if entity not in self._data:
+                    self._data[entity] = {"methods": []}
+
+                params = self._compile_operation_params(operation)
+                formatted_path = f"{http_method.upper()} {path}"
+                self._add_or_update_method(entity, method_name, formatted_path, params)
+
+    def scrape_content(
+        self, result: requests.Response | bytes | str | Any
+    ) -> dict[str, Any]:
```

Comment:
Don't proactively use type annotations
```suggestion
    def scrape_content(self, result):
```